diffrax/_solver/srk.py

```diff
@@ -0,0 +1,624 @@
+                else:
+                    assert self.bH_error is None
+                if self.bK is not None:
+                    assert self.bK_error is not None
+                    assert self.bK_error.shape == self.b_error.shape
+                else:
+                    assert self.bK_error is None
+
+        assert np.allclose(sum(self.b_sol), 1.0)
+
+        if self.ignore_stage_f is not None:
+            assert len(self.ignore_stage_f) == len(self.b_sol)
+        if self.ignore_stage_g is not None:
+            assert len(self.ignore_stage_g) == len(self.b_sol)
+
+
+StochasticButcherTableau.__init__.__doc__ = """**Arguments:**
+
+Let `k` denote the number of stages of the solver.
+
+- `a`: the lower triangle (without the diagonal) of the Butcher tableau. Should
+    be a tuple of NumPy arrays, corresponding to the rows of this lower triangle. The
+    first array represents the should be of shape `(1,)`. Each subsequent array should
+    be of shape `(2,)`, `(3,)` etc. The final array should have shape `(k - 1,)`.
+- `b_sol`: the linear combination of stages to take to produce the output at each step.
+    Should be a NumPy array of shape `(k,)`.
+- `b_error`: the linear combination of stages to take to produce the error estimate at
+    each step. Should be a NumPy array of shape `(k,)`. Note that this is *not*
+    differenced against `b_sol` prior to evaluation. (i.e. `b_error` gives the linear
+    combination for producing the error estimate directly, not for producing some
+    alternate solution that is compared against the main solution).
+- `c`: the time increments used in the Butcher tableau.
+    Should be a NumPy array of shape `(k-1,)`, as the first stage has time increment 0
+- `cW`: The coefficients in front of the Brownian increment at each stage.
+    Should be a NumPy array of shape `(k,)`. Only used when `additive_noise=True`.
+- `cH`: The coefficients in front of the space-time Lévy area at each stage.
+    Should be a NumPy array of shape `(k,)`. Only used when `additive_noise=True`.
+- `bW`: Coefficient for the Brownian increment when computing the
+    output $y_{n+1}$. Should be a `Scalar`.
+- `bH`: Coefficient for the space-time Lévy area when computing the
+    output $y_{n+1}$. Should be a `Scalar`.
+- `aW`: The coefficients in front of the Brownian increment at each stage.
+    Should be of the same shape as `a`. Only used when `additive_noise=False`.
+- `aH`: The coefficients in front of the space-time Lévy area at each stage.
+    Should be of the same shape as `a`. Only used when `additive_noise=False`.
+- `additive_noise`: Whether the SDE has additive noise. If `True`, then `bW` and `bH`
+    should be scalars, and `cW` and `cH` should be arrays of shape `(k,)`. If `False`,
+    then `bW` and `bH` should be arrays of shape `(k,)`, `cW` and `cH` should be `None`,
+    and `aW` and `aH` should be lists of arrays of the same shape as `a`.
```

Comment:
This is fairly complicated -- I think I'd probably advocate for the approach of making illegal types unrepresentable, something like:
```python
@dataclass(frozen=True)
class AbstractNoiseCoefficients:
    pass

@dataclass(frozen=True)
class AdditiveNoiseCoefficients(AbstractNoiseCoefficients):
    bW: FloatScalarLike
    bH: FloatScalarLike
    cW: Float[np.ndarray, "k"]
    cH: Float[np.ndarray, "k"]

@dataclass(frozen=True)
class VariableNoiseCoefficients(AbstractNoiseCoefficients):
    bW: Float[np.ndarray, "k"]
    bH: Float[np.ndarray, "k"]
    aW: tuple[np.ndarray, ...]
    aH: tuple[np.ndarray, ...]


@dataclass(frozen=True)
class StochasticButcherTableau:
    ...
    noise_coeffs: AbstractNoiseCoefficients
```

+1 for writing such clear documentation here that I was able to suggest the above, by the way!

Note that I've used `tuple` instead of `list`, here by the way -- this is really just a stylistic choice, but I quite like using immutable objects wherever possible.